### PR TITLE
Add DeprecatedBy and Source fields to annotations

### PR DESCRIPTION
**Pull request by nmittler**

### CODE REVIEW

File: cmd/annotations_prep/main.go [hunk 1]
```diff
@@ -91,12 +116,18 @@ type Instance struct {
...
+	// Source of this {{ .Collection.NameLowercase }}.
+	Source Source
+
 	// Hide the existence of this {{ .Collection.NameLowercase }} when outputting usage information.
 	Hidden bool
 
 	// Mark this {{ .Collection.NameLowercase }} as deprecated when generating usage information.
 	Deprecated bool
 
+	// A link to reference material on the replacement feature, if any.
```
Comment (louiscryan) [inline cmd/annotations_prep/main.go:128]: I think we have a standard for documentation relative links. Maybe just reference
https://preliminary.istio.io/latest/about/contribute/shortcodes/#add-links-to-other-pages
here.

File: cmd/annotations_prep/main.go [hunk 2]
```diff
@@ -61,6 +62,30 @@ func (s FeatureStatus) String() string {
 	return "Unknown"
 }
 
+// Source of the {{ .Collection.NameLowercase }}.
+type Source int
```
Comment (louiscryan) [inline cmd/annotations_prep/main.go:66]: Could there be another source beyond runtime & config? If not a 'runtime' boolean might do?

File: cmd/annotations_prep/main.go (hunk 1 above)
Comment (nmittler) [reply cmd/annotations_prep/main.go:128]: How do we document deprecation today? Do we have a separate page/heading for each? We would effectively need that in order for this to be useful.

File: cmd/annotations_prep/main.go [hunk 3]
```diff
@@ -61,6 +62,30 @@ func (s FeatureStatus) String() string {
...
+
+const (
+	// Config indicates the source of the {{ .Collection.NameLowercase }} is
+	// user configuration.
+	Config Source = iota
+
+	// Runtime indicates the source of the {{ .Collection.NameLowercase }} is
+	// the Istio runtime (i.e. generated by Istio).
+	Runtime
```
Comment (therealmitchconnors) [inline cmd/annotations_prep/main.go:75]: Many features are set to default values by Istio, but can be overridden by users.  In that case, is the Source Runtime or Config?

File: cmd/annotations_prep/main.go [hunk 4]
```diff
@@ -390,6 +432,9 @@ type Variable struct {
 	// Mark this annotation as deprecated when generating usage information.
 	Deprecated bool `json:"deprecated"`
 
+	// A link to reference material on the replacement feature, if any.
+	DeprecatedBy string
```
Comment (therealmitchconnors) [inline cmd/annotations_prep/main.go:436]: No json tag?

**Pull request closed without merging**